.github/workflows/release.yml

```diff
@@ -0,0 +1,33 @@
+# Publish Update python package at https://pypi.org/project/ceph-nvmeof/
+# This is official release, and will be pushed to pypi when a new tag is pushed in repo.
+# Usage: pip install ceph-nvmeof==1.2.15
+name: Upload release to PyPI
+on:
+  push:
+    tags:
+      - '*'
+    branches:
+      - 'devel'
+
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/ceph-nvmeof
+    if: startsWith(github.ref, 'refs/tags/')
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pdm-project/setup-pdm@v3
+        with:
+          python-version: 3.9
+          version: "2.7.4"
```

Comment:
I'd move all versions as input params of the workflow (defaulting to these same numbers), but that allows to easily build against newer/older Python/PDM versions.